src/Workspaces/SharedUtilitiesAndExtensions/Workspace/CSharp/LanguageServices/CSharpSyntaxFactsService.cs

```diff
@@ -2,1379 +2,26 @@
 // The .NET Foundation licenses this file to you under the MIT license.
 // See the LICENSE file in the project root for more information.
 
-using System;
-using System.Collections.Generic;
-using System.Collections.Immutable;
 using System.Diagnostics;
-using System.Diagnostics.CodeAnalysis;
 using System.Linq;
-using System.Text;
-using System.Threading;
 using Microsoft.CodeAnalysis.CSharp.Extensions;
-using Microsoft.CodeAnalysis.CSharp.Extensions.ContextQuery;
-using Microsoft.CodeAnalysis.CSharp.LanguageServices;
 using Microsoft.CodeAnalysis.CSharp.Syntax;
 using Microsoft.CodeAnalysis.Formatting;
 using Microsoft.CodeAnalysis.LanguageServices;
-using Microsoft.CodeAnalysis.PooledObjects;
-using Microsoft.CodeAnalysis.Shared.Extensions;
-using Microsoft.CodeAnalysis.Text;
-using Roslyn.Utilities;
 
 namespace Microsoft.CodeAnalysis.CSharp
 {
-    internal class CSharpSyntaxFactsService : AbstractSyntaxFactsService, ISyntaxFactsService
+    internal sealed class CSharpSyntaxFactsService : CSharpSyntaxFacts, ISyntaxFactsService
```

Comment:
In a follow-up PR I will make this type a private nested type of `CSharpSyntaxFactsServiceFactory`, so this will serve as a mere language service implementation to enable MEF based discovery. Subsequently, the code should become much cleaner:

1. All clients should operate with `ISyntaxFacts` for data query.
2. C# and VB specific singleton `Instance` should be accessed using `CSharpSyntaxFacts.Instance` and `VisualBasicSyntaxFacts.Instance`
3. `ISyntaxFactsService` should really only be used in context of `GetLanguageService<ISyntaxFactsService>` invocations as a pure discovery mechanism in language agnostic layer.
4. `CSharpSyntaxFactsService` and `VisualSyntaxFactsService` would not be accessible outside the corresponding language service factories.